platform/admin/admin-configuration-configinstaller/src/main/java/org/codice/ddf/admin/configuration/ConfigurationInstaller.java

```diff
@@ -13,204 +13,259 @@
-    AccessController.doPrivileged(
-        (PrivilegedAction<Void>)
-            () -> {
-              switch (event.getType()) {
-                case CM_UPDATED:
-                  LOGGER.debug("Handling update for pid {}", pid);
-                  handleUpdate(event);
-                  break;
-                case CM_DELETED:
-                  LOGGER.debug("Handling delete for pid {}", pid);
-                  handleDelete(event);
-                  break;
-                default:
-                  LOGGER.debug(
-                      "Unknown configuration event type, taking no action for pid {}", pid);
-              }
-              return null;
-            });
+            .map(configurationContextFactory::createContext)
+            .filter(context -> context.getConfigFile() != null)
+            .collect(Collectors.toMap(ConfigurationContext::getServicePid, CachedConfigData::new)));
   }
 
   /**
-   * Sync the config files in etc when an update occurs. There are four cases, which will be denoted
-   * using 2-tuple combinations of variables: [felixFileOriginal, felixFileNew]
+   * {@inheritDoc}
    *
-   * <p>[null, null] (1) Felix file prop wasn't tracked and is still not tracked. We don't need to
-   * do anything, and we further guarantee that beyond this point, one variable must not be null.
+   * <p>Sync the config files in etc when an update occurs. There are four cases, which will be
+   * denoted using 2-tuple combinations of variables: [felixFileFromCache, felixFileFromConfig]
    *
-   * <p>[null, X] (2) Unrecognized config with a felix file prop to track. The config's presence in
-   * etc was previously not being tracked by the system and we should start tracking it.
+   * <p>[null, null] (1) Config was created from config admin directly; not by dropping a file.
+   * There is nothing to track.
    *
-   * <p>[Y, null] [Y, X] (3) Felix prop was externally changed or removed. Revert this change.
+   * <p>[null, X] (2) Some config was just created by dropping a file into etc; track the file.
+   * Update the properties on disk if necessary and perform any special first time processing.
    *
-   * <p>[Y, Y] (4) Felix file prop did not change, do nothing. This is the happy path.
+   * <p>[Y, null] [Y, X] (3) Felix prop was externally changed or removed. This is an error.
    *
-   * @param event the configuration event for which an update is needed
+   * <p>[Y, Y] (4) Felix file prop did not change, do nothing. This is the happy path. Properties
+   * will be updated on disk if necessary.
    */
-  private void handleUpdate(ConfigurationEvent event) {
-    Configuration configuration = getConfiguration(event);
-    if (configuration == null) {
-      LOGGER.debug("Configuration was null for pid {}", event.getPid());
-      return;
-    }
+  @Override
+  public void handleStore(ConfigurationContext context) throws IOException {
+    String pid = context.getServicePid();
+    File fileFromConfig = context.getConfigFile();
```

Comment:
✏️  Could we call this `fileFromDisk`? It seems like an odd name, but the way the word `config` is used here is inconsistent with the way it's used on line 214 in the `writeIfNecessary` method.